pkg/ebpf/tracer_windows.go

```diff
@@ -2,23 +2,181 @@
+// _filterAddress defines an address to be matched, if supplied.
+// it can be ipv4 or ipv6 but not both.
+// supplying 0 for the address family means _any_ address (v4 or v6)
+	unsigned short af; //! AF_INET, AF_INET6 or 0
+	union
+	{
+		struct in6_addr         V6_address;
+		struct in_addr          V4_address;
+	}u;
+	unsigned short mask; // number of mask bits.
+} FILTER_ADDRESS;
+
+typedef enum
+{
+	DIRECTION_INBOUND = 0,
+	DIRECTION_OUTBOUND = 1
+} FILTER_DIRECTION;
+
+typedef struct _filterDefinition
+{
+	unsigned long Size;         //! size of this structure
+
+//  if supplied, the source and destination address must have the same address family.
+//  if both source and destination are applied, then the match for this filter
+//  is a logical AND, i.e. the source and destination both match.
+	unsigned short Af;     //! address family to filter
+
+	FILTER_ADDRESS  SourceAddress;
+	FILTER_ADDRESS  DestAddress;
+	unsigned short SourcePort;
+	unsigned short DestinationPort;
+	unsigned short Protocol;
+	FILTER_DIRECTION    Direction;
+} FILTER_DEFINITION;
+
+typedef struct _stats
+{
+    long Read_calls;		//! number of read calls to the driver
+    long Read_bytes;
+    long Read_calls_outstanding;
+    long Read_calls_completed;
+    long Read_calls_cancelled;
+    long Read_packets_skipped;
+    long Write_calls;	//! number of write calls to the driver
+    long Write_bytes;
+    long Ioctl_calls;	//! number of ioctl calls to the driver
+} STATS;
+
+typedef struct driver_stats
+{
+    STATS		Total;		//! stats since the driver was started
+    STATS		Handle;		//! stats for the file handle in question
+} DRIVER_STATS;
+*/
+import "C"
 import (
+	"fmt"
 	"github.com/DataDog/datadog-agent/pkg/process/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"syscall"
+	"unsafe"
+)
+
+const (
+	DRIVERFILE                            = "\\\\.\\ddfilter"
+	DDFILTER_IOCTL_GETSTATS               = 0x801
+	DDFILTER_IOCTL_SIMULATE_COMPLETE_READ = 0x802
+	DDFILTER_IOCTL_SET_FILTER             = 0x803
+	NETWORK_DEVICE_TYPE_CTL_CODE          = 0x00000012
+)
+
+var (
+	kernel32    = syscall.MustLoadDLL("kernel32.dll")
+	CreateFile  = kernel32.MustFindProc("CreateFileW")
+	CloseHandle = kernel32.MustFindProc("CloseHandle")
 )
 
 // Tracer struct for tracking network state and connections
 type Tracer struct {
-	config *Config
+	config       *Config
+	DriverHandle syscall.Handle
 }
 
 // NewTracer returns an initialized tracer struct
 func NewTracer(config *Config) (*Tracer, error) {
-	return &Tracer{}, nil
+	handle, err := openDriverFile(DRIVERFILE)
+	if err != nil {
+		return nil, fmt.Errorf("%s : %s", "Could not open driver file", err)
+	}
+	return &Tracer{
+		DriverHandle: handle,
+	}, nil
 }
 
 // Stop function stops running tracer
 func (t *Tracer) Stop() {}
 
+func openDriverFile(path string) (syscall.Handle, error) {
+	p, err := syscall.UTF16PtrFromString(path)
+	if err != nil {
+		return syscall.InvalidHandle, err
+	}
+	log.Info("Creating file...")
```

Comment:
I know these were for development, but we should log at most once in this function